pkg/declarativeconfig/notifier.go

```diff
@@ -0,0 +1,47 @@
+package declarativeconfig
+
+// KeyValuePair represents pair of key and value.
+type KeyValuePair struct {
+	Key   string `yaml:"key,omitempty"`
+	Value string `yaml:"value,omitempty"`
+}
+
+// GenericConfig is representation of storage.Notifier_Generic that supports transformation from YAML.
+type GenericConfig struct {
+	Endpoint            string         `yaml:"endpoint,omitempty"`
+	SkipTLSVerify       bool           `yaml:"skipTLSVerify,omitempty"`
+	CACert              string         `yaml:"caCert,omitempty"`
+	Username            string         `yaml:"username,omitempty"`
+	Password            string         `yaml:"password,omitempty"`
+	Headers             []KeyValuePair `yaml:"headers,omitempty"`
+	ExtraFields         []KeyValuePair `yaml:"extraFields,omitempty"`
```

Comment:
Wondering with these extra fields, do you have any examples of what those typically would be? Maybe it makes sense to create separate configs for the most prominent one instead of these key value pairs (thinking about this whether it could be the same as with the auth provider config which also was a map string string).

Reply:
Extra fields is something customer decides to send together with central's message:

```
Use Extra fields to include additional key-value pairs in the JSON object that Red Hat Advanced Cluster Security for Kubernetes sends. For example, if your webhook receiver accepts objects from multiple sources, you can add "source": "rhacs" as an extra field and then filter on this value to identify all alerts from Red Hat Advanced Cluster Security for Kubernetes.
```
So converting to a strict type is not possible here, it is list of key-value pairs by design.